zipline/pipeline/factors/statistical.py

```diff
@@ -475,3 +480,142 @@ def __new__(cls,
             regression_length=regression_length,
             mask=mask,
         )
+
+
+class SimpleBeta(CustomFactor, StandardOutputs):
+    """
+    Factor producing the slope of a regression line between each asset's daily
+    returns to the daily returns of a single "target" asset.
+
+    Parameters
+    ----------
+    target : zipline.Asset
+        Asset against which other assets should be regressed.
+    regression_length : int
+        Number of days of daily returns to use for the regression.
+    allowed_missing_percentage : float, optional
+        Percentage of returns observations that are allowed to be missing when
+        calculating betas. Default is 25%.
+    """
+    window_safe = True
+    dtype = float64_dtype
+    params = ('allowed_missing_count',)
+
+    @expect_types(
+        regression_length=int,
+        target=Asset,
+        allowed_missing_percentage=(int, float),
+        __funcname='SimpleBeta',
+    )
+    @expect_bounded(
+        regression_length=(3, None),
+        allowed_missing_percentage=(0.0, 1.0),
+        __funcname='SimpleBeta',
+    )
+    def __new__(cls,
+                target,
+                regression_length,
+                allowed_missing_percentage=0.25):
+        daily_returns = Returns(
+            window_length=2,
+            mask=(AssetExists() | SingleAsset(asset=target)),
+        )
+        allowed_missing_count = int(np.floor(
+            allowed_missing_percentage * regression_length
+        ))
+        return super(SimpleBeta, cls).__new__(
+            cls,
+            inputs=[daily_returns, daily_returns[target]],
+            window_length=regression_length,
+            allowed_missing_count=allowed_missing_count,
+        )
+
+    def compute(self,
+                today,
+                assets,
+                out,
+                all_returns,
+                target_returns,
+                allowed_missing_count):
+        vectorized_beta(
+            dependents=all_returns,
+            independent=target_returns,
+            allowed_missing=allowed_missing_count,
+            out=out,
+        )
+
+    def __repr__(self):
+        return "{}(window_length={}, allowed_missing={})".format(
+            type(self).__name__,
+            self.window_length,
+            self.params['allowed_missing'],
+        )
+
+
+def vectorized_beta(dependents, independent, allowed_missing, out=None):
```

Comment:
I couldn't find another place where `vectorized_beta` might be useful at the moment, however at a later date we might want to move it to `empyrical`.

Reply:
I don't think empyrical would use the `vectorized` part of this anywhere right now, but I suspect that the direct calculation of covariance and variance here is faster than what we currently do in empyrical, because it avoids making copies of the data and it avoids calculating parts of the covariance matrix that we don't need.  I'm not super worried about porting this in the near term, because I don't think anything is currently bottlenecked on empyrical's beta calculations, but if that becomes a bottleneck this is probably a good place to look to speed things up.